iodata/formats/fchk.py

```diff
@@ -435,3 +439,225 @@ def _triangle_to_dense(triangle: np.ndarray) -> np.ndarray:
+            coeffsb = data.mo.coeffsb[permutation] * signs.reshape(-1, 1)
+            _dump_real_arrays("Beta Orbital Energies", data.mo.energiesb, f)
+            _dump_real_arrays("Beta MO coefficients", coeffsb.transpose().flatten(), f)
+
+    # write reduced density matrix, if available
+    # get level of theory, use 'NA' if not available
+    level = data.lot.upper() if data.lot is not None else 'NA'
+    for item in ['MP2', 'MP3', 'CC', 'CI']:
+        if item in level:
+            level = item
+    for key, arr in data.one_rdms.items():
+        # get lower triangular elements of RDM
+        mat = arr[np.tril_indices(arr.shape[0])]
+
+        # identify type of RDMs
+        if key == "scf":
+            title = "Total SCF Density"
+        elif key == "scf_spin":
+            title = "Spin SCF Density"
+        elif key == "post_scf":
+            title = "Total {0} Density".format(level)
+        elif key == "post_scf_spin":
+            title = "Spin {0} Density".format(level)
+        else:
+            title = "Total SCF Density"
+        _dump_real_arrays(title, mat, f)
+
+    # write atomic charges
+    if 'mulliken' in data.atcharges:
+        _dump_real_arrays("Mulliken Charges", data.atcharges["mulliken"], f)
+    if 'esp' in data.atcharges:
+        _dump_real_arrays("ESP Charges", data.atcharges["esp"], f)
+    if 'npa' in data.atcharges:
+        _dump_real_arrays("NPA Charges", data.atcharges["npa"], f)
+
+    # write atomic gradient
+    if data.atgradient is not None:
+        _dump_real_arrays("Cartesian Gradient", data.atgradient.flatten(), f)
+
+    # write atomic hessian
+    if data.athessian is not None:
+        arr = data.athessian[np.tril_indices(data.athessian.shape[0])]
+        _dump_real_arrays("Cartesian Force Constants", arr, f)
+
+    # write moments
+    if (1, 'c') in data.moments:
+        _dump_real_arrays("Dipole Moment", data.moments[(1, 'c')], f)
+    if (2, 'c') in data.moments and len(data.moments[(2, 'c')]) != 0:
+        # quadrupole moments are stored as XX, XY, XZ, YY, YZ, ZZ in IOData, so they need to
+        # be permuted to have XX, YY, ZZ, XY, XZ, YZ order for FCHK.
+        quadrupole = data.moments[(2, 'c')][[0, 3, 5, 1, 2, 4]]
+        _dump_real_arrays("Quadrupole Moment", quadrupole, f)
+
+    # write polarizability tensor
+    if 'polarizability_tensor' in data.extra:
+        arr = data.extra["polarizability_tensor"]
+        arr = arr[np.tril_indices(arr.shape[0])]
+        _dump_real_arrays("Polarizability", arr, f)
+
+
+@document_dump_many(
+    "Gaussian Formatted Checkpoint",
+    ['atcoords', 'atnums', 'energy', 'mo', 'obasis'],
+    ['atcharges', 'atcorenums', 'atfrozen', 'atgradient', 'athessian', 'atmasses', 'lot',
+     'one_rdms', 'obasis_name', 'extra', 'moments'])
+def dump_many(f: TextIO, datas: Iterator[IOData]):
+    """Do not edit this docstring. It will be overwritten."""
+    for data in datas:
+        dump_one(f, data)
```

Comment:
This will not be compatible with `load_many`, which does not read a concatenation of FCHK files. It was specifically written to extract IRCs and (relaxed) scans from a single FCHK file as a list of IOData objects. I would suggest to ommit this function from the PR and open a separate PR with a `dump_many` that is compatible with `load_many`.

Reply:
I removed all related to `dump_many`